addons/mail/static/src/messaging/component/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -1014,6 +1017,72 @@ QUnit.test('open 3 different chat windows: not enough screen width', async funct
     );
 });
 
+QUnit.test('chat window: switch on TAB', async function (assert) {
+    assert.expect(7);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 1,
+                name: "channel1",
+            }, {
+                channel_type: "channel",
+                id: 2,
+                name: "channel2",
+            }],
+        },
+    });
+    await this.start();
+    await afterNextRender(
+        () => this.env.entities.Thread.find(thread => thread.id === 1).open()
+    );
+
+    assert.containsOnce(document.body, '.o_ChatWindow', 'Only one chatWindow must be opened');
+    assert.strictEqual(
+        document.querySelector('.o_ChatWindow .o_ComposerTextInput_textarea'),
+        document.activeElement,
+        "The chatWindow's composer must have focus"
+    );
+
+    await afterNextRender(() => {
+        triggerEvent(
+            document.querySelector('.o_ChatWindow .o_ComposerTextInput_textarea'),
+            'keydown',
+            {key: 'Tab'},
+        );
+    });
+    assert.strictEqual(
+        document.querySelector('.o_ChatWindow .o_ComposerTextInput_textarea'),
+        document.activeElement,
+        "The chatWindow's composer still has focus"
+    );
+
+    await afterNextRender(
+        () => this.env.entities.Thread.find(thread => thread.id === 2).open()
+    );
+    assert.containsN(document.body, '.o_ChatWindow', 2, 'Two chatWindows must be opened');
+    assert.strictEqual(
+        document.querySelectorAll('.o_ChatWindow .o_ComposerTextInput_textarea')[1],
+        document.activeElement,
+        "The second chatWindow's composer must have focus"
+    );
+
+    await afterNextRender(() => {
+        triggerEvent(
```

Comment:
`await` (this function is async, even if it comes from request animation frame I would play safe and do an await).

Or you could avoid putting an `await` if you use non-scoped arrow function (i.e. `(...args) => [...]` instead of `() => { [...] }`).